packages/jellyfish-transaction/__tests__/script/defi/dftx_token/TokenCreate.test.ts

```diff
@@ -0,0 +1,66 @@
+import { SmartBuffer } from 'smart-buffer'
+import { OP_DEFI_TX } from '../../../../src/script/defi'
+import { CTokenCreate, TokenCreate } from '../../../../src/script/defi/dftx_token'
+import { OP_CODES } from '../../../../src/script'
+import { toBuffer, toOPCodes } from '../../../../src/script/_buffer'
+import BigNumber from 'bignumber.js'
+
+/**
+ * using createToken sample from
+ * https://explorer.defichain.io/#/DFI/mainnet/tx/8a5066b4ea77c8d0b705ba94f47585f944ae587700f0f43f8655d01f38921f40
+ * https://explorer.defichain.io/#/DFI/mainnet/tx/baeddea27199a9e9001133f18942353dc79a765f0c437a1eda550f7675dc6b8b
+ * https://explorer.defichain.io/#/DFI/mainnet/tx/11fb90953bbd2a8f8649c116e6071dc73d428c5eba97c5a4a6dac550df2ab78c
+ */
+it('should bi-directional buffer-object-buffer', () => {
+  const fixtures = [
+    '6a1b44665478540342544307426974636f696e08000000000000000003',
+    '6a19446654785404474f4c4404476f6c6408000000000000000003',
+    '6a224466547854035753420e77616c6c7374726565746265747308000000000000000003'
+  ]
+
+  fixtures.forEach(hex => {
+    const stack: any = toOPCodes(
+      SmartBuffer.fromBuffer(Buffer.from(hex, 'hex'))
+    )
+    const buffer = toBuffer(stack)
+    expect(buffer.toString('hex')).toBe(hex)
+    expect((stack[1] as OP_DEFI_TX).tx.type).toBe(0x54)
+  })
+})
+
+const header = '6a224466547854' // OP_RETURN, PUSH_DATA(44665478, 54)
+const data = '035753420e77616c6c7374726565746265747308000000000000000003'
+const tokenCreate: TokenCreate = {
+  symbol: 'WSB',
+  name: 'wallstreetbets',
+  decimal: 8,
+  limit: new BigNumber('0'),
+  flags: 3
```

Comment:
Would be good to be able to set flags here, rather than a number. Should have some JS implementation of bitmask, can reference something like this: https://www.npmjs.com/package/bit-mask